apps/recovery-portal/src/main/webapp/error.jsp

```diff
@@ -87,6 +89,19 @@
 
     <script>
         function goBack() {
+    
+            var errorCodeFromParams = "<%=errorCode%>";
+            var invalidConfirmationErrorCode = "<%=IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_CODE.getCode()%>";
+
+            // Check if the error is related to the confirmation code being invalid.
+            // If so, navigate the users to the server home.
+            if (errorCodeFromParams === invalidConfirmationErrorCode) {
+                var redirect = "<%=IdentityUtil.getServerURL("", true, true)%>";
```

Comment:
I think we should redirect the user to reinitiate the self sign up flow. If this is directing the user to mgt console that might not be helpful. Can we redirect to user-portal self sign up in the product.? Thinking again it looks like in a separately hosted webpage case, we need this URL to be configurable too...

Reply:
I changed the code to redirect the users to the URL specified in the `callback` URL param.

Ex: In the following scenario, user will be directed to https://localhost:9443/registration.jsp when the `Go Back` button is clicked.

https://localhost:9443/accountrecoveryendpoint/confirmliteuserregistration.do?confirmation=2c4ea233-250c-4c33-8210-96a0f0725d02&tenantdomain=carbon.super&callback=https://localhost:9443/registration.jsp

As a precaution, i added a check to hide the `Go Back` button incase if the `callback` URL param is not available for some reason.